buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/header/InjectHeaderTests.java

```diff
@@ -9,344 +9,100 @@
-        YAMLParser yamlParser = YAML_FACTORY.createParser(testFile);
-        List<ObjectNode> tests = READER.<ObjectNode>readValues(yamlParser).readAll();
-        RestTestTransformer transformer = new RestTestTransformer();
-        // validate no setup
-        assertThat(tests.stream().filter(node -> node.get("setup") != null).count(), CoreMatchers.equalTo(0L));
-
-        List<ObjectNode> transformedTests = transformer.transformRestTests(
-            new LinkedList<>(tests),
-            Collections.singletonList(new InjectHeaders(headers))
-        );
+    public void testInjectHeadersNoPreExisting() throws Exception {
+        String testName = "/rest/transform/header/without_existing_headers.yml";
+        List<ObjectNode> tests = getTests(testName);
+        validateSetupDoesNotExist(tests);
+        List<ObjectNode> transformedTests = transformTests(tests);
         printTest(testName, transformedTests);
-        // ensure setup is correct
-        assertThat(transformedTests.stream().filter(node -> node.get("setup") != null).count(), CoreMatchers.equalTo(1L));
-        transformedTests.stream().filter(node -> node.get("setup") != null).forEach(this::assertSetup);
-        transformedTests.stream().filter(node -> node.get("teardown") != null).forEach(this::assertTeardown);
-        // ensure do body is correct
-        transformedTests.forEach(test -> {
-            Iterator<Map.Entry<String, JsonNode>> testsIterator = test.fields();
-            while (testsIterator.hasNext()) {
-                Map.Entry<String, JsonNode> testObject = testsIterator.next();
-                assertThat(testObject.getValue(), CoreMatchers.instanceOf(ArrayNode.class));
-                ArrayNode testBody = (ArrayNode) testObject.getValue();
-                assertTestBodyForHeaders(testBody, headers);
-            }
-        });
+        validateSetupAndTearDown(transformedTests);
+        validateBodyHasHeaders(transformedTests, headers);
     }
 
     /**
-     * test file has a setup: block, but no relevant children
+     * test file has preexisting headers
      */
     @Test
-    public void testInjectHeadersWithSetupBlock() throws Exception {
-        String testName = "/rest/transform/header/with_setup.yml";
-        File testFile = new File(getClass().getResource(testName).toURI());
-        YAMLParser yamlParser = YAML_FACTORY.createParser(testFile);
-        List<ObjectNode> tests = READER.<ObjectNode>readValues(yamlParser).readAll();
-        RestTestTransformer transformer = new RestTestTransformer();
-
-        // validate setup exists
-        assertThat(tests.stream().filter(node -> node.get("setup") != null).count(), CoreMatchers.equalTo(1L));
-
-        List<ObjectNode> transformedTests = transformer.transformRestTests(
-            new LinkedList<>(tests),
-            Collections.singletonList(new InjectHeaders(headers))
-        );
+    public void testInjectHeadersWithPreExisting() throws Exception {
```

Comment:
if the existing v7 test has a headers section (like for X-Opaque-ID), we should still add additional headers (Accept/Content-Type for instance)?
This test verifies that it should not happen?

Reply:
Actually this test tests that existing headers are preserved. In this case there is an existing header "foo" : "bar" and it is kept after injecting the headers. I had a minor non-logical setup w.r.t. the existence of the setup method, which i fixed. The output of the test is 
```
---
setup:
- skip:
    features: "headers"
---
Test without a setup:
- do:
    headers:
      foo: "bar"
      Accept: "application/vnd.elasticsearch+json;compatible-with=7"
      Content-Type: "application/vnd.elasticsearch+json;compatible-with=7"
    something:
      id: "something"
- match:
    acknowledged: true
```

lmk if you think that is incorrect, and I can fix on subsequent PR (trying to keep this PR refactoring only)

Reply:
makes sense! this is exactly what I was asking and it works as expected. thanks